consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/validation/MessageValidatorFactory.java

```diff
@@ -81,7 +81,7 @@ private SignedDataValidator createSignedDataValidator(
   public MessageValidator createMessageValidator(
       final ConsensusRoundIdentifier roundIdentifier, final BlockHeader parentHeader) {
     final BlockValidator blockValidator =
-        protocolSchedule.getByBlockNumber(roundIdentifier.getSequenceNumber()).getBlockValidator();
+        protocolSchedule.getByBlockHeader(parentHeader).getBlockValidator();
```

Comment:
Similarly to the IbftRoundFactory this needs to get the block validator based on the block we are building represented by the sequence number, not the parent block.